apps/vetrina/src/Vetrina/Main.purs

```diff
@@ -231,7 +239,10 @@ render self =
             , children: [ DOM.text "OK" ]
             }
         ]
-    PurchaseUnexpectedError -> DOM.text "SOMETHING WENT HORRIBLY WRONG SERVER SIDE"
+    PurchaseUnexpectedError -> vetrinaContainer $ Array.singleton $
+      Purchase.Error.error
+        { onRetry: self.setState \_ -> initialState
```

Comment:
You cannot simply set it to `initialState`, as for instance the loading spinner is then switched on (`isLoading: Just Spinner.Loading)`. Also it's wiping all the state we actually need (sentry logger, user, product selection).

Would it be enough to set the `purchaseState` to `NewPurchase`?

Reply:
This is what we're doing when the user cancels the process (from NETS terminal) https://github.com/KSF-Media/affresco/blob/master/apps/vetrina/src/Vetrina/Main.purs#L178-L180, and the action should be same here.

Although, I noticed this is not enough, as we are not recognizing the user we just created. The view should be the new-purchase-with-logged-in-user one.

Reply:
We should also set the `accountStatus` depending on what the `state.user` is, like here https://github.com/KSF-Media/affresco/blob/master/apps/vetrina/src/Vetrina/Main.purs#L124-L129

Reply:
It seems to be working, but what about the `newOrder`, `OrderFailure`, `PaymentMethod`? Don't we need to wipe them if we are starting anew?

Reply:
`paymentMethod` is always `CreditCard` for now, so that's fine. However, I was looking at `newOrder` and `orderFailure` of `state`, and thinking do we even use those anywhere (except for _setting_ them)? 😄 If not, just remove them from `state`.